### ISSUE TITLE
Bug: no-async-array-callbacks doesn't recognize Promise.race/Promise.any

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#27]
## Description

The `no-async-array-callbacks` rule only recognizes `Promise.all` and `Promise.allSettled` as valid consumers of async map results. It doesn't recognize `Promise.race` or `Promise.any`, which are also valid patterns:

```ts
// This should be allowed but is flagged
const result = await Promise.race(items.map(async (item) => fetchItem(item)));

// This should also be allowed
const first = await Promise.any(sources.map(async (s) => trySource(s)));
```

## Suggested Fix

Add `race` and `any` to the list of recognized Promise methods in `isPromiseAllCall`.
